content/references/rippled-api/api-conventions/rate-limiting.md

```diff
@@ -0,0 +1,62 @@
+# Rate Limiting
+
+The `rippled` server limits the rate at which API clients can make requests on public APIs. Rate limiting is based on the IP address of the client, so clients behind [network address translation](https://en.wikipedia.org/wiki/Network_address_translation) share a limit based on their public IP address.
+
+**Tip:** Rate limiting does not apply when the client is connected [as an admin](get-started-with-the-rippled-api.html#admin-access).
+
+When a client is approaching the rate limit, the server adds the field `"warning": "load"` at the top level of an [API response](response-formatting.html). This warning is not added to every response, but the server may send several such warnings before it disconnects a client.
+
+If a client goes past the rate limit, the server disconnects that client and does not serve further requests from the client's API address for a while. The disconnect message is different for the WebSocket and JSON-RPC APIs.
+
+## WebSocket API Disconnect Message
+
+For a WebSocket API connection, the server closes the connection and provides a close message and code with the close message. The way you access these messages depends on your WebSocket client implementation. For example, using the [Node.js ws library](https://github.com/websockets/ws), the following code prints the close reason when disconnected:
+
+```js
+const WebSocket = require('ws')
+const ws = new WebSocket('ws://localhost:6007/')
+ws.on('close', (code,reason) => {
+  console.log("Disconnected. \ncode: ", code, "\nreason: ", reason)
+})
+
+// If rate limited, prints:
+// Disconnected.
+// code:  1008
+// reason:  threshold exceeded
+```
+
+If the connection is closed because of rate limiting, the close code is `1008` and the reason is the string `threshold exceeded`.
```

Comment:
the close code is `1008`, and the close message is the string `threshold exceeded`.